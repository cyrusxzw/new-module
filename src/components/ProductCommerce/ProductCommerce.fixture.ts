const ProductCommerceFixture = {
  cta: {
    text: 'Lorem ipsum dolor',
    url: '/',
  },
  copy: {
    size: 'Size',
  },
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse et ornare mauris, id dictum ante. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Mauris volutpat molestie lobortis. Nunc vel aliquam odio.',
  eyebrow: 'Vestibulum ante ipsum primis',
  heading: 'Lorem ipsum dolor ',
  image: {
    altText: 'alt-tag',
    sizes: {
      large: './assets/images/Product/variant-one-large.png',
      medium: './assets/images/Product/variant-one-medium.png',
      small: './assets/images/Product/variant-one-small.png',
    },
  },
  theme: 'dark',
  variantOptions: [
    {
      size: '50 mL - In stock',
      sku: 'ARD33',
      price: '$26.45',
      isInStock: true,
      image: {
        altText: 'alt-tag-ARD33',
        sizes: {
          large: './assets/images/Product/variant-one-large.png',
          medium: './assets/images/Product/variant-one-medium.png',
          small: './assets/images/Product/variant-one-small.png',
        },
      },
    },
    {
      size: '100 mL - Not in stock',
      sku: 'ARD32',
      price: '$86.97',
      isInStock: false,
      image: {
        altText: 'alt-tag-ARD32',
        sizes: {
          large: './assets/images/Product/variant-two-large.png',
          medium: './assets/images/Product/variant-two-medium.png',
          small: './assets/images/Product/variant-two-small.png',
        },
      },
    },
    {
      size: '180 mL - In stock',
      sku: 'ARD30',
      price: '$96.97',
      isInStock: true,
      image: {
        altText: 'alt-tag-ARD32',
        sizes: {
          large: './assets/images/Product/variant-two-large.png',
          medium: './assets/images/Product/variant-two-medium.png',
          small: './assets/images/Product/variant-two-small.png',
        },
      },
    },
    {
      alternateAction: {
        url: '/?searchMenu=open',
        label: 'Visit Store - $89.45',
      },
      size: '500 mL - Alternate action',
      sku: 'ARD31',
      price: '$109.45',
      isInStock: true,
      image: {
        altText: 'alt-tag-ARD31',
        sizes: {
          large: './assets/images/Product/variant-two-large.png',
          medium: './assets/images/Product/variant-two-medium.png',
          small: './assets/images/Product/variant-two-small.png',
        },
      },
    },
  ],
};

export { ProductCommerceFixture };
