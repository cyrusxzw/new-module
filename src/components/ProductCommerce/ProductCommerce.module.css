@import 'styles/index';

.base {
  font-family: var(--font-family-sans-serif-regular);
  font-size: rem(14px);
  font-weight: var(--font-weight-normal);

  @media (--viewport-xs-sm-only) {
    padding-right: 20px;
    padding-left: 20px;
  }

  @media (--viewport-md) {
    display: grid;
    grid-template-columns: repeat(12, 1fr);
    grid-template-rows: repeat(8, min-content);
  }

  &,
  & * {
    box-sizing: border-box;
  }
}

.imageWrapper {
  width: 100%;
  margin-bottom: 10px;

  @media (--viewport-md) {
    display: flex;
    padding-right: 0;
    padding-left: 34px;
    margin-bottom: 0;
    grid-column: 1/ 5;
    grid-row: 1/ -1;
  }

  @media (--viewport-lg) {
    padding-left: calc(var(--layout-lg-spacing) * 2);
  }
}

.image {
  display: flex;
  align-items: flex-end;
  justify-content: center;

  @media (--viewport-md) {
    justify-content: flex-end;
  }

  img {
    width: auto;
    max-width: 100%;
    max-height: 300px;

    @media (--viewport-md) {
      max-height: 400px;
    }
  }
}

.description {
  margin-bottom: 20px;
  font-size: rem(14px);

  & > :last-child {
    margin-bottom: 0;
  }
}

.variantsWrapper {
  margin-bottom: 30px;
}

.variants {
  justify-content: flex-start;
}

.linkButtonGroup {
  @media (--viewport-lg) {
    display: grid;
    grid-template-columns: repeat(2, minmax(0, 1fr));
  }
}

.addToCartButton,
.cta {
  max-width: none;
}

/* Content Padding */
.header,
.description,
.variantsWrapper,
.linkButtonGroup {
  max-width: 600px;
  @media (--viewport-md-only) {
    padding-right: 34px;
  }
  @media (--viewport-md) {
    grid-column: 6/ 13;
  }

  @media (--viewport-lg) {
    grid-column: 6/ 11;
  }
}

@media all and (-ms-high-contrast: none) {
  .imageWrapper {
    -ms-grid-column-span: 4;
    grid-column-start: 1;
    grid-row-end: 8;
    -ms-grid-row-span: 8;
    grid-row-start: 1;

    img {
      flex-shrink: 0;
    }
  }
  .description {
    grid-row-start: 2;
  }
  .variantsWrapper {
    grid-row-start: 3;
  }
  .linkButtonGroup {
    grid-row-start: 4;
  }

  .header,
  .description,
  .variantsWrapper,
  .linkButtonGroup {
    -ms-grid-column-span: 5;
    grid-column-start: 6;
  }
}
