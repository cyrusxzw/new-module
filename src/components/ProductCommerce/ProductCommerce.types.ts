import type { Themes } from '~/types';

type ProductCommerceProps = {
  className?: string;
  copy?: {
    addToCart?: {
      cartAction?: string;
      updateNotification?: string;
      outOfStock?: {
        label?: string;
        title?: string;
      };
    };
    size?: string;
  };
  cta?: {
    text?: string;
    url?: string;
  };
  description?: React.ReactNode;
  eyebrow?: string;
  heading?: string;
  id?: string;
  isActive?: boolean;
  onCtaClick?: () => void;
  size?: string;
  theme?: Themes;
};

export type { ProductCommerceProps };
