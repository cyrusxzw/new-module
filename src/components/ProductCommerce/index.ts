import { ProductCommerce } from './ProductCommerce';

export { ProductCommerce };
