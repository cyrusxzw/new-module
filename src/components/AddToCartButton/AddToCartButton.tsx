import React from 'react';
import cx from 'classnames';
import { useAddToCartContext, useVariantSelectContext } from '~/contexts';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import { Button } from '~/components/Button';
import { Loading } from '~/components/Loading';
import { Hyperlink } from '~/components/Hyperlink';
import { useDelayedDisabled } from './AddToCartButton.utils';
import type { AddToCartButtonType } from './AddToCartButton.types';
import styles from './AddToCartButton.module.css';

const AddToCartButton: AddToCartButtonType = ({
  className,
  copy,
  dataTestRef = 'ADD_TO_CART',
  isEnabled = true,
  isFullWidth = true,
  theme = 'dark',
}) => {
  const addToCartContext = useAddToCartContext();
  const { selectedVariant } = useVariantSelectContext();
  const {
    errorMessage,
    hasError,
    isLoading,
    isUpdateSuccessful,
  } = addToCartContext;
  const shouldShowUpdateSuccessMessage = !isLoading && isUpdateSuccessful;
  const isDelayedDisabled = useDelayedDisabled(shouldShowUpdateSuccessMessage);
  const updateNotificationLabel = copy?.updateNotification;

  if (!selectedVariant) return null;

  const {
    alternateAction,
    isInStock,
    isSellable,
    price,
    sku,
  } = selectedVariant;

  const cartActionText = copy?.cartAction
    ? `${copy?.cartAction} â€” ${price}`
    : price;
  const postTaxPriceText = copy?.postTaxPrice ? ` ${copy?.postTaxPrice}` : '';
  const cartActionLabel = cartActionText + postTaxPriceText;

  const classSet = cx(
    styles.base,
    { [styles.fullWidth]: isFullWidth },
    className,
  );

  const handleOnClick = () => {
    const { dispatch, onClick } = addToCartContext;

    onClick(sku, dispatch);
  };

  if (!!alternateAction?.url && !!alternateAction?.label) {
    return (
      <Hyperlink
        className={classSet}
        dataTestRef={dataTestRef}
        hasTargetInNewWindow={alternateAction?.openInANewWindow}
        isAlternate={true}
        style={HYPERLINK_STYLE_TYPES.INTERNAL_NO_ICON_BUTTON_LINK}
        theme={theme}
        title={alternateAction?.label}
        url={alternateAction?.url}
      >
        {alternateAction?.label}
      </Hyperlink>
    );
  }

  if (hasError) {
    /** @TODO Handle errors thrown by handleOnClick */
    console.error('Add To Cart button updateError: ', errorMessage); // eslint-disable-line
  }

  const isButtonEnabled =
    !isLoading &&
    !!price &&
    !!sku &&
    isInStock &&
    !!sku &&
    isEnabled &&
    !hasError &&
    !isDelayedDisabled &&
    isSellable;

  const labelClassName = cx(
    styles.label,
    { [styles.hideLabel]: isLoading },
    { [styles.showSuccessMessage]: shouldShowUpdateSuccessMessage },
  );

  return (
    <Button
      className={classSet}
      dataTestRef={dataTestRef}
      isAlternate={true}
      isEnabled={isButtonEnabled}
      onClick={handleOnClick}
      theme={theme}
      title={isInStock ? cartActionLabel : copy?.outOfStock?.title}
    >
      {isLoading && (
        <Loading
          className={styles.loading}
          data-test-ref={`${dataTestRef}_UPDATING`}
          isLoading={true}
          theme={theme === 'dark' ? 'light' : 'dark'}
        />
      )}
      <span
        className={labelClassName}
        data-test-ref={
          shouldShowUpdateSuccessMessage ? `${dataTestRef}_SUCCESS` : undefined
        }
      >
        {shouldShowUpdateSuccessMessage ? (
          <span>{updateNotificationLabel}</span>
        ) : (
          <span> </span>
        )}
        <span>{isInStock ? cartActionLabel : copy?.outOfStock?.label}</span>
      </span>
    </Button>
  );
};

export { AddToCartButton };
