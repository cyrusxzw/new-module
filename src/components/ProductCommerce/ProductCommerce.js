import React, { useRef } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import { HEADING, HYPERLINK_STYLE_TYPES, TRANSITIONS } from '~/constants';
import { useProductDetailContext, useVariantSelectContext } from '~/contexts';
import { useImageTransition } from '~/customHooks';
import { getVariantRadioOptions } from '~/utils/product';
import { AddToCartButton } from '~/components/AddToCartButton';
import { Heading } from '~/components/Heading';
import { Hyperlink } from '~/components/Hyperlink';
import { Image } from '~/components/Image';
import { LinkButtonGroup } from '~/components/LinkButtonGroup';
import { Loading } from '~/components/Loading';
import { RadioGroup } from '~/components/RadioGroup';
import { SectionHeading } from '~/components/SectionHeading';
import { Transition } from '~/components/Transition';

import styles from './ProductCommerce.module.css';

const ProductCommerce = React.forwardRef(
  (
    {
      className,
      copy,
      cta,
      description,
      eyebrow,
      heading,
      id,
      size,
      theme,
      isActive,
      onCtaClick,
    },
    ref,
  ) => {
    const imageRef = useRef();
    const {
      selectedVariant,
      onVariantChange,
      variants,
    } = useVariantSelectContext();

    const isBundle = variants?.length === 0;

    const { productDetail } = useProductDetailContext();
    const { sku } = productDetail;

    const [currentImage, isImageActive] = useImageTransition(
      selectedVariant?.image || productDetail.image,
      imageRef,
    );

    if (!isBundle && !selectedVariant) {
      return <Loading isLoading={true} />;
    }

    const variantRadioOptions = getVariantRadioOptions(variants);
    const classSet = cx(styles.base, className);
    const RADIO_GROUP_NAME = sku;
    const RADIO_GROUP_DATA_TEST_REF = 'PRODUCT_COMMERCE_VARIANT_SELECT';

    return (
      <div className={classSet} id={id} ref={ref}>
        <Transition isActive={isActive} type={TRANSITIONS.TYPE.SLOW_FADE}>
          <SectionHeading
            className={styles.header}
            eyebrow={eyebrow}
            heading={heading}
            isFlush={true}
            theme={theme}
          />
        </Transition>
        <Transition isActive={isActive} type={TRANSITIONS.TYPE.SLOW_FADE}>
          <div className={styles.description}>{description}</div>
        </Transition>
        <Transition isActive={isActive} type={TRANSITIONS.TYPE.SHIFT_IN_LEFT}>
          <div className={styles.imageWrapper}>
            <Transition isActive={isImageActive} type={TRANSITIONS.TYPE.FADE}>
              <Image
                altText={currentImage.altText}
                className={styles.image}
                large={currentImage.sizes?.large}
                medium={currentImage.sizes?.medium}
                ref={imageRef}
                small={currentImage.sizes?.small}
              />
            </Transition>
          </div>
        </Transition>
        {size && !isBundle && (
          <Transition isActive={isActive} type={TRANSITIONS.TYPE.SLOW_FADE}>
            <div className={styles.variantsWrapper}>
              <Heading
                hasMediumWeightFont={true}
                isFlush={true}
                level={HEADING.LEVEL.FOUR}
                size={HEADING.SIZE.X_X_SMALL}
                theme={theme}
              >
                {copy?.size}
              </Heading>
              <RadioGroup
                className={styles.variants}
                dataTestRef={RADIO_GROUP_DATA_TEST_REF}
                name={RADIO_GROUP_NAME}
                onChange={e => onVariantChange(e, variants)}
                options={variantRadioOptions}
                theme={theme}
                value={selectedVariant.sku}
              />
            </div>
          </Transition>
        )}

        <Transition isActive={isActive} type={TRANSITIONS.TYPE.SHIFT_IN_DOWN}>
          <LinkButtonGroup
            className={styles.linkButtonGroup}
            hasFitContent={false}
            isFlush={false}
            textAlign="center"
            theme={theme}
          >
            <AddToCartButton
              className={styles.addToCartButton}
              copy={copy.addToCart}
              isFullWidth={false}
              theme={theme}
            />
            {cta && (
              <Hyperlink
                className={styles.cta}
                isAlternate={false}
                onClick={() => onCtaClick()}
                style={HYPERLINK_STYLE_TYPES.INTERNAL_BUTTON_LINK}
                url={cta.url}
              >
                {cta.text}
              </Hyperlink>
            )}
          </LinkButtonGroup>
        </Transition>
      </div>
    );
  },
);

ProductCommerce.propTypes = {
  className: PropTypes.string,
  onCtaClick: PropTypes.func,
  copy: PropTypes.shape({
    addToCart: PropTypes.shape({
      cartAction: PropTypes.string,
      updateNotification: PropTypes.string,
      outOfStock: PropTypes.shape({
        label: PropTypes.string,
        title: PropTypes.string,
      }),
    }),
    size: PropTypes.string,
  }),
  cta: PropTypes.shape({
    text: PropTypes.string,
    url: PropTypes.string,
  }),
  description: PropTypes.element,
  eyebrow: PropTypes.string,
  heading: PropTypes.string,
  id: PropTypes.string,
  size: PropTypes.string,
  isActive: PropTypes.bool,
  theme: PropTypes.oneOf(['dark', 'light']),
};

ProductCommerce.defaultProps = {
  className: undefined,
  copy: {
    addToCart: {
      cartAction: undefined,
      updateNotification: undefined,
      outOfStock: {
        label: undefined,
        title: undefined,
      },
    },
    size: undefined,
  },
  cta: undefined,
  description: undefined,
  eyebrow: undefined,
  heading: undefined,
  id: undefined,
  onCtaClick: () => {},
  size: undefined,
  isActive: true,
  theme: 'dark',
};

export { ProductCommerce };
