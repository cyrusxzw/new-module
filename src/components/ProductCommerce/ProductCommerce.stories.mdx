import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import { StorybookGridOverlay } from '~.storybook/components';
import {
  AddToCartContextProvider,
  ProductDetailContextProvider,
  VariantSelectContextProvider,
} from '~/contexts';
import { Podium } from '~/components/Podium';
import { P } from '~/components/Paragraph';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import { Hyperlink } from '~/components/Hyperlink';
import { AddToCartButtonFixture } from '~/components/AddToCartButton/AddToCartButton.fixture';
import { mockAddToCartButtonOnClick } from '~/components/AddToCartButton/__mocks__/AddToCartButton.onClick';
import { ProductDetailHeaderFixture } from '~/components/ProductDetailHeader/ProductDetailHeader.fixture';
import { ProductCommerce } from './ProductCommerce';
import { ProductCommerceFixture } from './ProductCommerce.fixture';

<Meta title="Components/ProductCommerce" component={ProductCommerce} />

# ProductCommerce

<Canvas>
  <Story name="Base component">
    {(() => {
      const cta = knobs.object('cta', ProductCommerceFixture.cta, 'Content');
      const eyebrow = knobs.text(
        'eyebrow',
        ProductCommerceFixture.eyebrow,
        'Content',
      );
      const heading = knobs.text(
        'heading',
        ProductCommerceFixture.heading,
        'Content',
      );
      const isActive = knobs.boolean(
        'isActive',
        true,
        'Presentation',
      );
      const theme = knobs.select(
        'theme',
        ['dark', 'light'],
        'dark',
        'Presentation',
      );
      const implementation = {
        gridOverlay: {
          isActive: knobs.boolean('Show grid overlay', false, 'Implementation'),
          hasInvertedColours: knobs.boolean(
            'Invert grid overlay colours',
            false,
            'Implementation',
          ),
        },
      };
      return (
        <>
        <Podium verticalPadding="small">
          <AddToCartContextProvider onClick={mockAddToCartButtonOnClick}>
            <ProductDetailContextProvider
              product={ProductDetailHeaderFixture.product}
            >
              <VariantSelectContextProvider
                variants={ProductCommerceFixture.variantOptions}
              >
                <ProductCommerce
                  isActive={isActive}
                  copy={{
                    addToCart: AddToCartButtonFixture.copy,
                    ...ProductCommerceFixture.copy,
                  }}
                  cta={cta}
                  description={
                    <P>
                      {knobs.text(
                        'description',
                        ProductCommerceFixture.description,
                        'Content',
                      )}
                    </P>
                  }
                  eyebrow={eyebrow}
                  heading={heading}
                  size={ProductDetailHeaderFixture.product.size}
                  theme={theme}
                />
              </VariantSelectContextProvider>
            </ProductDetailContextProvider>
          </AddToCartContextProvider>
        </Podium>
          <StorybookGridOverlay
            isActive={implementation.gridOverlay.isActive}
            hasInvertedColours={implementation.gridOverlay.hasInvertedColours}
          />
        </>
      );
    })()}
  </Story>
</Canvas>

## API

<ArgsTable of={ProductCommerce} />
