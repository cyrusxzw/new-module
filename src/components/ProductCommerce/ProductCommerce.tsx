import React, { forwardRef, useRef } from 'react';
import cx from 'classnames';
import { useThemeContext } from '~/contexts';
import { HYPERLINK_STYLE_TYPES } from '~/constants';
import { useProductDetailContext, useVariantSelectContext } from '~/contexts';
import { useImageTransition } from '~/customHooks';
import { getVariantRadioOptions } from '~/utils/product';
import { AddToCartButton } from '~/components/AddToCartButton';
import { Heading } from '~/components/Heading/index.ts';
import { Hyperlink } from '~/components/Hyperlink';
import { Image } from '~/components/Image';
import { LinkButtonGroup } from '~/components/LinkButtonGroup/index.ts';
import { Loading } from '~/components/Loading';
import { RadioGroup } from '~/components/RadioGroup/index.ts';
import { SectionHeading } from '~/components/SectionHeading/index.ts';
import { Transition } from '~/components/Transition';
import type { ProductCommerceProps } from './ProductCommerce.types';
import styles from './ProductCommerce.module.css';

const ProductCommerce = forwardRef<HTMLDivElement, ProductCommerceProps>(
  (
    {
      className,
      copy,
      cta,
      description,
      eyebrow,
      heading,
      id,
      size,
      theme,
      isActive = true,
      onCtaClick,
    },
    ref,
  ) => {
    const imageRef = useRef();
    const currentTheme = useThemeContext(theme, 'dark');
    const {
      selectedVariant,
      onVariantChange,
      variants,
    } = useVariantSelectContext();
    const { productDetail } = useProductDetailContext();
    const [currentImage, isImageActive] = useImageTransition(
      selectedVariant?.image || productDetail.image,
      imageRef,
    );

    const isBundle = variants?.length === 0;

    if (!isBundle && !selectedVariant) {
      return <Loading isLoading={true} />;
    }

    const { sku } = productDetail;
    const handleOnCtaClick = () => onCtaClick();
    const handleOnRadioGroupChange = (
      event: React.ChangeEvent<HTMLInputElement>,
    ) => onVariantChange(event, variants);
    const variantRadioOptions = getVariantRadioOptions(variants);
    const classSet = cx(styles.base, className);
    const RADIO_GROUP_NAME = sku;
    const RADIO_GROUP_DATA_TEST_REF = 'PRODUCT_COMMERCE_VARIANT_SELECT';

    return (
      <div className={classSet} id={id} ref={ref}>
        <Transition isActive={isActive} type="slowFade">
          <SectionHeading
            className={styles.header}
            eyebrow={eyebrow}
            heading={heading}
            isFlush={true}
            theme={currentTheme}
          />
        </Transition>
        <Transition isActive={isActive} type="slowFade">
          <div className={styles.description}>{description}</div>
        </Transition>
        <Transition isActive={isActive} type="shiftInLeft">
          <div className={styles.imageWrapper}>
            <Transition isActive={isImageActive} type="fade">
              <Image
                altText={currentImage.altText}
                className={styles.image}
                ref={imageRef}
                sizes={currentImage.sizes}
              />
            </Transition>
          </div>
        </Transition>
        {size && !isBundle && (
          <Transition isActive={isActive} type="slowFade">
            <div className={styles.variantsWrapper}>
              <Heading
                hasMediumWeightFont={true}
                isFlush={true}
                level="4"
                size="xXSmall"
                theme={currentTheme}
              >
                {copy?.size}
              </Heading>
              <RadioGroup
                className={styles.variants}
                dataTestRef={RADIO_GROUP_DATA_TEST_REF}
                name={RADIO_GROUP_NAME}
                onChange={handleOnRadioGroupChange}
                options={variantRadioOptions}
                theme={currentTheme}
                value={selectedVariant.sku}
              />
            </div>
          </Transition>
        )}

        <Transition isActive={isActive} type="shiftInDown">
          <LinkButtonGroup
            className={styles.linkButtonGroup}
            hasFitContent={false}
            isFlush={false}
            textAlign="center"
            theme={currentTheme}
          >
            <AddToCartButton
              className={styles.addToCartButton}
              copy={copy?.addToCart}
              dataTestRef="product-commerce"
              isFullWidth={false}
              theme={currentTheme}
            />
            {cta && (
              <Hyperlink
                className={styles.cta}
                isAlternate={false}
                onClick={handleOnCtaClick}
                style={HYPERLINK_STYLE_TYPES.INTERNAL_BUTTON_LINK}
                url={cta.url}
              >
                {cta.text}
              </Hyperlink>
            )}
          </LinkButtonGroup>
        </Transition>
      </div>
    );
  },
);

export { ProductCommerce };
