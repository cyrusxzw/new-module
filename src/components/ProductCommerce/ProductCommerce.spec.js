import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import {
  AddToCartContextProvider,
  ProductDetailContextProvider,
  VariantSelectContextProvider,
} from '~/contexts';
import { ProductDetailHeaderFixture } from '~/components/ProductDetailHeader/ProductDetailHeader.fixture';
import { AddToCartButtonFixture } from '~/components/AddToCartButton/AddToCartButton.fixture';
import { mockAddToCartButtonOnClick } from '~/components/AddToCartButton/__mocks__/AddToCartButton.onClick';
import { ProductCommerce } from './ProductCommerce';
import { ProductCommerceFixture } from './ProductCommerce.fixture';

describe('<ProductCommerce />', () => {
  it('should render base component correctly', () => {
    const product = ProductDetailHeaderFixture.product;
    const variants = ProductDetailHeaderFixture.product.variantOptions;

    const { container } = render(
      <AddToCartContextProvider onClick={mockAddToCartButtonOnClick}>
        <ProductDetailContextProvider product={product}>
          <VariantSelectContextProvider variants={variants}>
            <ProductCommerce
              copy={{
                addToCart: AddToCartButtonFixture.copy,
                ...ProductCommerceFixture.copy,
              }}
              cta={ProductCommerceFixture.cta}
              description={<p>ProductCommerceFixture.description</p>}
              eyebrow={ProductCommerceFixture.eyebrow}
              heading={ProductCommerceFixture.heading}
              onAddToCartClick={mockAddToCartButtonOnClick}
              productName={ProductCommerceFixture.productName}
              theme="dark"
              variants={ProductCommerceFixture.variantOptions}
            />
          </VariantSelectContextProvider>
        </ProductDetailContextProvider>
      </AddToCartContextProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should execute cta click correctly', () => {
    const product = ProductDetailHeaderFixture.product;
    const variants = ProductDetailHeaderFixture.product.variantOptions;

    const onCtaClickMock = jest.fn();

    render(
      <AddToCartContextProvider onClick={mockAddToCartButtonOnClick}>
        <ProductDetailContextProvider product={product}>
          <VariantSelectContextProvider variants={variants}>
            <ProductCommerce
              copy={{
                addToCart: AddToCartButtonFixture.copy,
                ...ProductCommerceFixture.copy,
              }}
              cta={ProductCommerceFixture.cta}
              description={<p>ProductCommerceFixture.description</p>}
              eyebrow={ProductCommerceFixture.eyebrow}
              heading={ProductCommerceFixture.heading}
              onAddToCartClick={mockAddToCartButtonOnClick}
              onCtaClick={onCtaClickMock}
              productName={ProductCommerceFixture.productName}
              theme="dark"
              variants={ProductCommerceFixture.variantOptions}
            />
          </VariantSelectContextProvider>
        </ProductDetailContextProvider>
      </AddToCartContextProvider>,
    );

    const link = screen.getAllByRole('link')[0]; // First Hyperlink with click function
    userEvent.click(link);

    expect(onCtaClickMock).toHaveBeenCalledTimes(1);
  });
});
