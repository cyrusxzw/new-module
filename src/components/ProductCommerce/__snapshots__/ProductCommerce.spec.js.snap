// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCommerce /> should render base component correctly 1`] = `
<div>
  <div
    class="base"
  >
    <header
      class="header base"
      data-testid="data-testid-SectionHeading"
    >
      <h2
        class="base flush xLarge dark heading"
      >
        Lorem ipsum dolor 
      </h2>
      <h3
        class="base flush mediumWeightFont xXSmall dark eyebrow"
      >
        Vestibulum ante ipsum primis
      </h3>
    </header>
    <div
      class="description"
    >
      <p>
        ProductCommerceFixture.description
      </p>
    </div>
    <div
      class="imageWrapper"
    >
      <picture
        class="base picture dark image fadeEnter"
      >
        <img
          loading="lazy"
        />
      </picture>
    </div>
    <div
      class="base linkButtonGroup"
      data-testid="data-testid-LinkButtonGroup"
    >
      <button
        class="base alternate blockStyle dark base addToCartButton link center"
        data-test-ref="ADD_TO_CART"
        title="Add to your cart — $26.45"
        type="button"
      >
        <span
          class="label"
        >
          <span>
             
          </span>
          <span>
            Add to your cart — $26.45
          </span>
        </span>
      </button>
      <a
        class="base blockStyle hasIcon center dark cta link center"
        href="/"
        target="_self"
      >
        Lorem ipsum dolor
         
        <i
          aria-hidden="true"
          class="icon"
        >
          <svg
            aria-labelledby="rightArrow-1"
            class="base rightArrow dark"
            data-testid="data-testid-Icon"
            focusable="false"
            height="12"
            role="img"
            style="height: 12px; width: 12px;"
            tabindex="-1"
            viewBox="0 0 50 50"
            width="12"
          >
            <title
              id="rightArrow-1"
            >
              rightArrow
            </title>
            <g>
              <path
                d="M30.1,5.3L50,25.1L30.1,45h-6.6l18-17.6H0v-4.8h41.5l-18-17.6h6.6V5.3z"
              />
            </g>
          </svg>
        </i>
      </a>
    </div>
  </div>
</div>
`;
